modules/player/src/main/java/net/hollowcube/player/PlayerImpl.java

```diff
@@ -94,6 +104,37 @@ private void handleDiggingPacket(ClientPlayerDiggingPacket packet) {
         }
     }
 
+    private void handlePositionPacket(ClientPlayerPositionPacket packet) {
+        if (packet.onGround()) {
+
+        }
+        else {
+            if (this.getInstance().getBlock(this.getPosition()) == Block.BUBBLE_COLUMN) {
+                //TODO: I think we want to make this part of a whole PlayerInWaterHandler or something, and on top of that,
+                // we would have to make a BlockComponentHandler or something to encapsulate it. I wonder if @Matt has a
+                // suggestion on this.
+
+                // PlayerBubbleColumnSinkEvent playerBubbleColumnSinkEvent = new PlayerBubbleColumnSinkEvent(this, this.getFood());
+                // MinecraftServer.getGlobalEventHandler().call(playerBubbleColumnSinkEvent);
+
+                if (!this.wasLastSeenInBubbleColumn) {
+                    this.wasLastSeenInBubbleColumn = true;
+                    this.bubbleColumnSinkEnergyLevel = this.getFood();
+                    //TODO: This does not work lol it is not implemented in minestom to get rid of the player's
+                    // ability to sprint when they have hunger below 7
+                    this.setFood(6);
+                }
+            }
+            else if (this.wasLastSeenInBubbleColumn) {
+                if (this.getInstance().getBlock(this.getPosition()) != Block.BUBBLE_COLUMN) {
```

Comment:
This separate check can be discarded, as you check it in the first if statement, and if we reach this else if, we know we are not in a bubble column.